### PR TITLE
Reduce PR testing matrix.

### PR DESCRIPTION
**Pull request by alliepiper**
1. Temporarily remove current nightly matrix pending NVKS bringup.
2. Move current per-PR matrix to nightly.
3. Reduce the number of jobs in the PR matrix while maintaining decent coverage.

Before: (total jobs: 437)

| Runners | |
|------|------ |
|  320 | `linux-amd64-cpu16` |
|   66 | `linux-amd64-gpu-v100-latest-1` |
|   28 | `linux-arm64-cpu16` |
|   23 | `windows-amd64-cpu16` |

After (total jobs: 183)

| Runners | |
|------|------
|  126 | `linux-amd64-cpu16`
|   26 | `linux-amd64-gpu-v100-latest-1`
|   21 | `windows-amd64-cpu16`
|   10 | `linux-arm64-cpu16`

### CODE REVIEW

File: ci/matrix.yaml [hunk 1]
```diff
@@ -11,6 +11,49 @@ workflows:
...
+    - {jobs: ['test_lid1',  'test_lid2'], project: ['cub'], std: 'max',    cxx: ['gcc']}
+    - {jobs: ['test_nolid', 'test_lid0'], project: ['cub'], std: 'max',    cxx: ['clang', 'msvc']}
+
+    # Modded builds:
+    - {jobs: ['build'], std: 'max', cxx: ['gcc', 'clang'], cpu: 'arm64'}
+    - {jobs: ['build'], std: 'max', cxx: ['gcc'], sm: '90a'}
+    # default_projects: clang-cuda
+    - {jobs: ['build'], std: 'max', cudacxx: 'clang', cxx: 'clang'}
+    # nvrtc:
+    - {jobs: ['nvrtc'], project: 'libcudacxx', std: 'minmax'}
```
Comment (miscco) [inline ci/matrix.yaml:33]: There is only one nvrtc job so I would really like full coverage here
```suggestion
    - {jobs: ['nvrtc'], project: 'libcudacxx', std: 'all'}
```

File: ci/matrix.yaml [hunk 2]
```diff
@@ -11,6 +11,49 @@ workflows:
 
   pull_request:
     # Old CTK
+    - {jobs: ['build'], std: 'all', ctk: '11.1', cxx: ['gcc6', 'clang9', 'msvc2017']}
+    - {jobs: ['build'], std: 'all', ctk: '11.8', cxx: ['gcc11'], sm: '60;70;80;90'}
+    # Current CTK build-only
+    - {jobs: ['build'], std: 'all', cxx: ['gcc7', 'clang9', 'msvc2019']}
+    - {jobs: ['build'], std: 'all', cxx: ['gcc', 'clang', 'msvc', 'intel']}
```
Comment (jrhemstad) [inline ci/matrix.yaml:18]: question: What does it mean when a compiler name like `gcc` is specified without a version?

Comment (alliepiper) [reply ci/matrix.yaml:18]: It's a shortcut to always use the latest version.

File: ci/matrix.yaml [hunk 3]
```diff
@@ -11,6 +11,49 @@ workflows:
...
+    # Split up cub tests:
+    - {jobs: ['test_nolid', 'test_lid0'], project: ['cub'], std: 'minmax', cxx: ['gcc']}
+    - {jobs: ['test_lid1',  'test_lid2'], project: ['cub'], std: 'max',    cxx: ['gcc']}
+    - {jobs: ['test_nolid', 'test_lid0'], project: ['cub'], std: 'max',    cxx: ['clang', 'msvc']}
+
+    # Modded builds:
+    - {jobs: ['build'], std: 'max', cxx: ['gcc', 'clang'], cpu: 'arm64'}
+    - {jobs: ['build'], std: 'max', cxx: ['gcc'], sm: '90a'}
+    # default_projects: clang-cuda
+    - {jobs: ['build'], std: 'max', cudacxx: 'clang', cxx: 'clang'}
```
[the remark this answers is not in the document]
Comment (alliepiper) [reply ci/matrix.yaml:31]: We can def do this, since it won't increase our GPU runner count 👍

File: ci/matrix.yaml (hunk 1 above)
Comment (alliepiper) [reply ci/matrix.yaml:33]: This will cost a few more GPU jobs, but seems reasonable.

**Review by miscco — APPROVED** [review]

**Comment by bdice** [comment]
Reducing `linux-amd64-gpu-v100-latest-1` usage is the main consideration for me. If the additional CPU jobs provide greater confidence in build correctness, feel free to leave more of those in place. I don’t think we are contending for CPU as much as GPU.

**Review by bdice — APPROVED** [review]
Thanks for this reduction!

**Comment by jrhemstad** [comment]
In the spirit of minimum viable changes for the immediate need, I'd prefer to just reduce the number of jobs that use the GPU runners and keep the build jobs unchanged.

**Pull request merged**